assets/js/modules/tagmanager/hooks/useExistingTagEffect.test.js

```diff
@@ -99,4 +99,189 @@ describe( 'useExistingTagEffect', () => {
 			true
 		);
 	} );
+
+	it( 'does not change the useSnippet value when there is already a container ID on page load (container ID is same as existing tag)', async () => {
+		const account = factories.accountBuilder();
+		// eslint-disable-next-line sitekit/acronym-case
+		const accountID = account.accountId;
+		const containers = factories.buildContainers(
+			2,
+			// eslint-disable-next-line sitekit/acronym-case
+			{ accountId: account.accountId, usageContext: [ CONTEXT_WEB ] }
+		);
+		const [ existingContainer, anotherContainer ] = containers;
+
+		registry
+			.dispatch( MODULES_TAGMANAGER )
+			// eslint-disable-next-line sitekit/acronym-case
+			.receiveGetExistingTag( existingContainer.publicId );
+
+		registry
+			.dispatch( MODULES_TAGMANAGER )
+			.receiveGetAccounts( [ account ] );
+		registry
+			.dispatch( MODULES_TAGMANAGER )
+			.receiveGetContainers( containers, { accountID } );
+		registry.dispatch( MODULES_TAGMANAGER ).setAccountID( accountID );
+
+		// Manually enable useSnippet.
+		registry.dispatch( MODULES_TAGMANAGER ).setUseSnippet( true );
+
+		registry
+			.dispatch( MODULES_TAGMANAGER )
+			// eslint-disable-next-line sitekit/acronym-case
+			.setContainerID( existingContainer.publicId );
+		registry.dispatch( MODULES_TAGMANAGER ).setInternalContainerID(
+			// eslint-disable-next-line sitekit/acronym-case
+			existingContainer.containerId
+		);
+
+		const { rerender } = renderHook( () => useExistingTagEffect(), {
+			registry,
+		} );
+
+		expect( registry.select( MODULES_TAGMANAGER ).getUseSnippet() ).toBe(
+			true
+		);
+
+		act( () => {
+			// Set useSnippet to false to simulate pressing the toggle.
+			registry.dispatch( MODULES_TAGMANAGER ).setUseSnippet( false );
+			// Change to another container. This should not change the useSnippet value.
```

Comment:
Seeing as the snippet has just been set to `true`, and is tested to be `false` at the next `expect`, this comment is a little confusing, maybe the "not" should be removed?

```suggestion
			// Change to existing tag container. This should change the useSnippet value.
```

Reply:
+1. Updated.